File: scripts/01_load_data.R
library(tidyverse)
library(arrow)
library(janitor)
library(lubridate)

cat("Loading data...\n")

encounters <- open_dataset("data/encounters.csv", format = "csv")

patients <- read_csv("data/patients.csv", col_types = cols(
  DurableKey = col_integer(),
  CensusBlockGroupFipsCode = col_character(),
  FirstRace = col_character(),
  MaritalStatus = col_character(),
  MyChartStatus = col_character(),
  OmbEthnicity = col_character(),
  OmbRace = col_character(),
  SexAssignedAtBirth = col_character(),
  SexualOrientation = col_character(),
  SmokingStatus = col_character(),
  VitalStatus = col_character(),
  PatientBirthYearBin = col_integer()
), show_col_types = FALSE) |>
  clean_names()

diagnosis <- read_csv("data/diagnosis.csv", col_types = cols(
  DiagnosisKey = col_integer(),
  DiagnosisValue = col_character(),
  GroupCode = col_character()
), show_col_types = FALSE) |>
  clean_names()

departments <- read_csv("data/departments.csv", show_col_types = FALSE) |>
  clean_names()

providers <- read_csv("data/providers.csv", show_col_types = FALSE) |>
  clean_names()

social_determinants <- open_dataset("data/social_determinants.csv", format = "csv")

tigercensuscodes <- read_csv("data/tigercensuscodes.csv", show_col_types = FALSE) |>
  clean_names()

cat("\nSummary of loaded data:\n")
cat("encounters:           ", nrow(encounters |> select(1) |> collect()), "rows\n")
cat("patients:             ", nrow(patients), "rows\n")
cat("diagnosis:            ", nrow(diagnosis), "rows\n")
cat("departments:          ", nrow(departments), "rows\n")
cat("providers:            ", nrow(providers), "rows\n")
cat("social_determinants:  ", nrow(social_determinants |> select(1) |> collect()), "rows\n")
cat("tigercensuscodes:     ", nrow(tigercensuscodes), "rows\n")

dir.create("outputs", showWarnings = FALSE)
saveRDS(list(
  patients = patients,
  diagnosis = diagnosis,
  departments = departments,
  providers = providers
), "outputs/reference_tables.rds")

cat("\nReference tables saved to outputs/reference_tables.rds\n")

File: scripts/01_load_data.py
import pandas as pd
import os

print("Loading data...")

encounters = pd.read_csv("data/encounters.csv")
patients = pd.read_csv("data/patients.csv")
diagnosis = pd.read_csv("data/diagnosis.csv")
departments = pd.read_csv("data/departments.csv")
providers = pd.read_csv("data/providers.csv")
social_determinants = pd.read_csv("data/social_determinants.csv")
tigercensuscodes = pd.read_csv("data/tigercensuscodes.csv")

for df in [encounters, patients, diagnosis, departments, providers, social_determinants, tigercensuscodes]:
    df.columns = [col.lower() for col in df.columns]

print("\nSummary of loaded data:")
print(f"encounters:           {len(encounters)} rows")
print(f"patients:             {len(patients)} rows")
print(f"diagnosis:            {len(diagnosis)} rows")
print(f"departments:          {len(departments)} rows")
print(f"providers:            {len(providers)} rows")
print(f"social_determinants:  {len(social_determinants)} rows")
print(f"tigercensuscodes:     {len(tigercensuscodes)} rows")

os.makedirs("outputs", exist_ok=True)

patients.to_pickle("outputs/patients.pkl")
diagnosis.to_pickle("outputs/diagnosis.pkl")
departments.to_pickle("outputs/departments.pkl")
providers.to_pickle("outputs/providers.pkl")

print("\nReference tables saved to outputs/")
